src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -0,0 +1,337 @@
+    }
+
+    log.warn(LOG_AREA, message);
+}
+
+function buildConnection({ baseUrl, contextId, authToken, protocol }) {
+    const accessTokenFactory = () => {
+        return authToken.replace('BEARER ', '');
+    };
+    const url = `${baseUrl}/oapi/streaming?contextId=${contextId}`;
+
+    return new window.signalrCore.HubConnectionBuilder()
+        .withUrl(url, {
+            accessTokenFactory,
+        })
+        .withHubProtocol(protocol)
+        .configureLogging({
+            log: handleLog,
+        })
+        .build();
+}
+
+/*
+ * There are few inconsistencies while using different protocols
+ * It normalizes the messages received for different protocols
+ */
+function normalizeMessage(message, protocol) {
+    // JSON protocol use camel casing and message pack use pascal case
+    const referenceId = message.referenceId || message.ReferenceId;
+    let dataFormat = message.payloadFormat || message.PayloadFormat;
+    let data = message.payload || message.Payload;
+
+    // JSON format. Normalized to old streaming format for backward compatibility
+    // Message pack parses the enum and converts it to string
+    if (dataFormat === 1 || dataFormat === 'Json') {
+        dataFormat = 0;
+    }
+
+    // Protobuf format. Normalized to old streaming format for backward compatibility
+    if (dataFormat === 2 || dataFormat === 'Protobuf') {
+        dataFormat = 1;
+    }
+
+    // JSON protocol converts bytes array to base64 encoded string
+    if (protocol.name === 'json' && typeof data === 'string') {
+        data = new Uint8Array(
+            window
+                .atob(data)
+                .split('')
+                .map((char) => char.charCodeAt(0)),
+        );
+    }
+
+    return {
+        ReferenceId: referenceId,
+        DataFormat: dataFormat,
+        Data: data,
+    };
+}
+
+function parseMessage(message, utf8Decoder) {
+    const { ReferenceId, DataFormat } = message;
+    let data = message.Data;
+
+    // JSON
+    if (DataFormat === 0) {
+        try {
+            data = utf8Decoder.decode(data);
+            data = JSON.parse(data);
+        } catch (e) {
+            const error = new Error(e.message);
+            error.payload = data;
+
+            throw error;
+        }
+    }
+
+    return {
+        ReferenceId,
+        DataFormat,
+        Data: data,
+    };
+}
+
+function SignalrCoreTransport(baseUrl, transportFailCallback = NOOP) {
+    this.name = streamingTransports.SIGNALR_CORE;
+    this.baseUrl = baseUrl;
+    this.connection = null;
+    this.authToken = null;
+    this.contextId = null;
+    this.messageStream = null;
+    this.hasStreamingStarted = false;
+    this.isStopping = false;
+
+    // callbacks
+    this.transportFailCallback = transportFailCallback;
+    this.stateChangedCallback = NOOP;
+    this.receivedCallback = NOOP;
+    this.errorCallback = NOOP;
+    this.unauthorizedCallback = NOOP;
+
+    try {
+        this.utf8Decoder = new window.TextDecoder();

```

Comment:
Theres no callback for reconnecting - if thats handled by signal-r core itself, any idea how long it is?

I think you should test scenarios where you remove internet on both phone and desktop - the saxotrader code relies on getting a state event in order to show the offline notification.
In addition the orphan finder may start resetting subscriptions (but that should be ok since it now copes with network errors)

Reply:
its configurable 
I wanted to have discussion around it on slack since we already have reconnect logic in streaming

Reply:
its fine if a transport does its own reconnecting, espescially in this case, we are happy if it doesn't fallback to anything else.. but we need to notify outside that we are reconnecting so we show a network error dialog.

Reply:
I have added reconnect delays with exponential increment 
please check `RECONNECT_DELAYS` constant at the top